### DISCUSSION TITLE
Can I convert xsh to py?

### DISCUSSION THREAD

**Discussion by fabceolin** [xonsh/xonsh#4236] [Q&A]
Is it possible to convert a xsh script to python script? I want to use a pyinstaller to generate a static executable (of couse, the shell commands binaries will not be inside the executable)

**Comment by anki-code** [comment]
Generally yes. You can rewrite any xonsh script to Python using [xonsh API](https://xon.sh/api/index.html). 

For example
```python
# xonsh
$PATH
# EnvPath([
# '/usr/bin'
# ])

# xonsh API
from xonsh.environ import Env
env = Env()
env.get('PATH')
# EnvPath([
# '/usr/bin'
# ])
```
